storage/storage.go

```diff
@@ -0,0 +1,331 @@
+		login, password).Scan(&cnt)
+	if err != nil {
+		return err
+	}
+	if cnt != 1 {
+		return ErrInvalidLoginOrPassword
+	}
+	return nil
+}
+
+func (db *DatabaseInstance) GetAuth(userID string) (login string, err error) {
+	ctx := context.Background()
+	var loginRes string
+	err = db.conn.QueryRow(ctx, "SELECT login FROM market.authentication where cookie = $1;",
+		userID).Scan(&loginRes)
+	if err != nil && errors.Is(err, sql.ErrNoRows) {
+		return "", err
+	}
+	return loginRes, nil
+}
+
+func (db *DatabaseInstance) InsertOrder(orderID string, status string, accrual float32, login string) error {
+	ctx := context.Background()
+
+	_, err := db.conn.Exec(ctx, "INSERT INTO market.orders (orderID, status , accrual, login , uploadDateTime) "+
+		"VALUES ($1, $2, $3, $4, $5) ON CONFLICT (orderID) DO UPDATE SET status = excluded.status, "+
+		"accrual = excluded.accrual;",
+		orderID, status, accrual, login, time.Now())
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (db *DatabaseInstance) GetLoginByOrderID(orderID string) (string, error) {
+	ctx := context.Background()
+	var login string
+
+	err := db.conn.QueryRow(ctx, "SELECT login FROM market.orders where orderID = $1;",
+		orderID).Scan(&login)
+
+	if err != nil && errors.Is(err, sql.ErrNoRows) {
+		return "", err
+	}
+	if len(login) != 0 {
+		return login, nil
+	}
+	return "", nil
+}
+
+func (db *DatabaseInstance) RecountBalanceForLogin(login string) error {
+	ctx := context.Background()
+	var sum float32
+	var withdrawSum float32 = 0
+	err := db.conn.QueryRow(ctx, "SELECT sum(accrual) FROM market.orders where login = $1;",
+		login).Scan(&sum)
+	if err != nil && !strings.Contains(err.Error(), "can't scan into dest") {
+		return err
+	}
+
+	err = db.conn.QueryRow(ctx, "SELECT sum(sum) FROM market.withdrawals where login = $1;",
+		login).Scan(&withdrawSum)
+	if err != nil && !strings.Contains(err.Error(), "can't scan into dest") {
+		log.Println()
+		return err
+	}
+
+	sum = sum - withdrawSum
+
+	_, err = db.conn.Exec(ctx, "INSERT INTO market.balance (login, balance, withdrawn) "+
+		"VALUES ($1, $2, $3) ON CONFLICT (login) DO UPDATE SET balance = excluded.balance,"+
+		"  withdrawn = excluded.withdrawn;",
+		login, sum, withdrawSum)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (db *DatabaseInstance) GetAllOrdersOfUser(login string) ([]OrderInfo, error) {
+	ctx := context.Background()
+	var results []OrderInfo
+	rows, err := db.conn.Query(ctx, "SELECT orderID, status, accrual, uploadDateTime "+
+		"FROM market.orders WHERE login = $1", login)
+	if err != nil {
+		return nil, err
+	}
+	i := 0
+	for rows.Next() {
+		var timeFromRow time.Time
+		var accrual float32
+		var number string
+		var status string
+		err := rows.Scan(&number, &status, &accrual, &timeFromRow)
+		if err != nil {
+			continue
+		}
+		results = append(results, OrderInfo{Number: number, Status: status})
+		if accrual != 0 {
+			results[i].Accrual = accrual
+		}
```

Comment:
А мы так не можем сделать? Если accrual == 0, то results[i].Accrual по дефолту тоже 0, мне кажется условие лишним
```suggestion
		results = append(results, OrderInfo{Number: number, Status: status, Accrual: accrual})
```